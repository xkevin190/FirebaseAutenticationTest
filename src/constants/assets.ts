export const FONTS = {
    LATO: 'Lato',
};