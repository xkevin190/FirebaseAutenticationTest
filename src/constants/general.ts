export const SCREEN_ANIMATION = 'slide_from_right';

export const APIS = {
  USERS: {
    GET_IN_TOUCH: 'users/getInTouch',
  },
  AUTH: {
    CREATE_ACCOUNT: 'auth/parents/signup',
    LOGIN: 'auth/parents/login',
    RECOVER_PASSWORD: 'auth/recover',
  },
};

export const NO_CONNECTION_ERROR = 'TypeError: Network request failed';
