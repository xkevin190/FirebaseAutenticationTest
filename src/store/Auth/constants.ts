export type IAccountActionTypes = {
  CREATE_ACCOUNT: string;
  LOGIN: string;
  ClOSE_SESSION: string;
};

export const AccounthActionsTypes: IAccountActionTypes = {
  CREATE_ACCOUNT: 'CREATE_ACCOUNT',
  LOGIN: ' LOGIN',
  ClOSE_SESSION: 'ClOSE_SESSION',
};
