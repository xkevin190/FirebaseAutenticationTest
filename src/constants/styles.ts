import { FONTS } from "./assets";

export const COLORS = {
    WHITE: '#FFFFFF',
    BLACK: '#000000',
    BLACK_025: '#00000040',
    GRAY_DEFAULT: '#46484B',
    GRAY_DEFAULT_DARKER: '#646565',
    GRAY_MEDIUM: '#9DA5B9',
    GRAY_LIGHT: '#F6F8FF',
    GRAY_LIGHT_DARKER: '#DDDDDD',
    DISABLED_DEFAULT: '#DFE6ED',
    OPYA_BLUE: '#2162E3',
    OPYA_VIOLET: '#1745D5',
    MAIN_DEFAULT: '#313342',
    SECONDARY_LIGHT: '#62698B',
    ALERT_DEFAULT: '#D15686',
    ALERT_LIGHT: '#FFD0E2',
  };

  const FONTS_TYPES = {
    BLACK: 'Black',
    BLACK_ITALIC: 'BlackItalic',
    BOLD: 'Bold',
    BOLD_ITALIC: 'BoldItalic',
    HAIR_LINE: 'HairLine',
    HAIR_LINE_ITALIC: 'HairLineItalic',
    HEAVY: 'Heavy',
    HEAVY_ITALIC: 'HeavyItalic',
    ITALIC: 'Italic',
    LIGHT: 'Light',
    LIGHT_ITALIC: 'LightItalic',
    MEDIUM: 'Medium',
    MEDIUM_ITALIC: 'MediumItalic',
    REGULAR: 'Regular',
    SEMI_BOLD: 'SemiBold',
    SEMI_BOLD_ITALIC: 'SemiBoldItalic',
    THIN: 'Thin',
    THIN_ITALIC: 'ThinItalic',
  };


  export const STYLES = {
    FONT_FAMILY: {
      LATO_REGULAR: `${FONTS.LATO}-${FONTS_TYPES.REGULAR}`,
      LATO_BOLD: `${FONTS.LATO}-${FONTS_TYPES.BOLD}`,
    },
    FONT_SIZE: {
      PARAGRAPH: 14,
      SMALL_PARAGRAPH: 12,
      MEDIUM_PARAGRAPH: 16,
      INTERMEDIUM_PARAGRAPH: 18,
      LARGE_PARAGRAPH: 20,
    },
  };
  

