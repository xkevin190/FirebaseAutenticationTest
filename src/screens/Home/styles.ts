import {StyleSheet, ViewStyle} from 'react-native';

import {COLORS} from '../../constants/styles';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    flexDirection: 'column',
    backgroundColor: COLORS.GRAY_LIGHT,
  } as ViewStyle,
});

export default styles;
