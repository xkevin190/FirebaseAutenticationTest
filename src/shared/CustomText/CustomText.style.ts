import {StyleSheet, TextStyle} from 'react-native';

import {STYLES} from '../../constants/styles';

const styles = StyleSheet.create({
  'small-paragraph-bold': {
    fontFamily: STYLES.FONT_FAMILY.LATO_BOLD,
    fontSize: STYLES.FONT_SIZE.SMALL_PARAGRAPH,
  } as TextStyle,
  'small-paragraph': {
    fontFamily: STYLES.FONT_FAMILY.LATO_REGULAR,
    fontSize: STYLES.FONT_SIZE.SMALL_PARAGRAPH,
  } as TextStyle,
  'medium-paragraph': {
    fontFamily: STYLES.FONT_FAMILY.LATO_REGULAR,
    fontSize: STYLES.FONT_SIZE.MEDIUM_PARAGRAPH,
  } as TextStyle,
  paragraph: {
    fontFamily: STYLES.FONT_FAMILY.LATO_REGULAR,
    fontSize: STYLES.FONT_SIZE.PARAGRAPH,
  } as TextStyle,
  'large-paragraph': {
    fontFamily: STYLES.FONT_FAMILY.LATO_REGULAR,
    fontSize: STYLES.FONT_SIZE.LARGE_PARAGRAPH,
  } as TextStyle,
  'paragraph-bold': {
    fontFamily: STYLES.FONT_FAMILY.LATO_BOLD,
    fontSize: STYLES.FONT_SIZE.PARAGRAPH,
  } as TextStyle,
  'intermedium-paragraph-bold': {
    fontFamily: STYLES.FONT_FAMILY.LATO_BOLD,
    fontSize: STYLES.FONT_SIZE.INTERMEDIUM_PARAGRAPH,
  } as TextStyle,
  'intermedium-paragraph': {
    fontSize: STYLES.FONT_SIZE.INTERMEDIUM_PARAGRAPH,
    fontFamily: STYLES.FONT_FAMILY.LATO_REGULAR,
  } as TextStyle,
});

export default styles;
