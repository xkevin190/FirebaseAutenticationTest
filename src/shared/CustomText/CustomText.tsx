import React from 'react';
import {Text, TextStyle} from 'react-native';

import {COLORS} from '../../constants/styles';
import {TextBody} from '../../types/CustomText';
import styles from './CustomText.style';

export interface ICustomTextProps {
  text: string;
  center?: boolean;
  color?: string;
  bodyType?: TextBody;
  testID?: string;
  customStyle?: TextStyle;
}

const CustomText: React.FC<ICustomTextProps> = ({
  text,
  color = COLORS.BLACK,
  bodyType = 'paragraph',
  center = false,
  testID = '',
  customStyle = {},
}) => (
  <Text
    style={{
      ...styles[bodyType],
      ...customStyle,
      ...(center && {textAlign: 'center'}),
      color,
    }}
    testID={testID}>
    {text}
  </Text>
);

export default CustomText;
