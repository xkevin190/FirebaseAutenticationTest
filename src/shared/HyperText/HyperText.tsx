import React from 'react';
import { TouchableOpacity, Text, ViewStyle } from 'react-native';

import { COLORS } from '../../constants/styles';
import CustomText from '../CustomText';

export interface IHyperTextProps {
  prefix?: string;
  prefixColor?: string;
  text: string;
  textColor?: string;
  testID: string;
  customStyles?: ViewStyle;
  onPress: () => void;
}

const HyperText: React.FC<IHyperTextProps> = ({
  prefix,
  prefixColor = COLORS.GRAY_MEDIUM,
  text,
  textColor = COLORS.OPYA_BLUE,
  testID,
  customStyles,
  onPress,
}) => (
  <TouchableOpacity onPress={onPress} testID={testID} style={customStyles}>
    <Text>
      {prefix && <CustomText color={prefixColor} text={`${prefix} `} />}
      <CustomText color={textColor} text={text} bodyType="paragraph-bold" />
    </Text>
  </TouchableOpacity>
);

export default HyperText;
