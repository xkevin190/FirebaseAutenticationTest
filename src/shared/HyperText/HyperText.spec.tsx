import React from 'react';
import {shallow} from 'enzyme';
import {TouchableOpacityProps} from 'react-native';

import {COLORS} from '../../constants/styles';
import {ICustomTextProps} from '../CustomText/CustomText';
import HyperText, {IHyperTextProps} from './HyperText';

describe('HyperText (Unit-Test)', () => {
  const props: IHyperTextProps = {
    testID: 'TEST_ID',
    onPress: jest.fn(),
    text: 'TEST_TEXT',
  };
  let wrapper = shallow(<HyperText {...props} />);

  describe('general functionality', () => {
    it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should render default button with "testID" prop', () => {
      const buttonProps = wrapper
        .find('TouchableOpacity')
        .props() as TouchableOpacityProps;
      expect(buttonProps.testID).toEqual(props.testID);
    });

    it('should call "onPress" prop when tapped', () => {
      const button = wrapper.find('TouchableOpacity');
      button.simulate('press');
      expect(props.onPress).toBeCalledTimes(1);
    });

    it('should render "text" prop', () => {
      const textProps = wrapper.find('CustomText').props() as ICustomTextProps;
      expect(textProps.text).toEqual(props.text);
      expect(textProps.color).toEqual(COLORS.OPYA_BLUE);
    });
  });

  describe('when optional props are received', () => {
    it('should render "text" with "textColor" prop received', () => {
      wrapper = shallow(
        <HyperText {...props} textColor={COLORS.GRAY_DEFAULT} />,
      );
      const textProps = wrapper.find('CustomText').props() as ICustomTextProps;
      expect(textProps.color).toEqual(COLORS.GRAY_DEFAULT);
    });

    it('should render prefix text when "prefix" prop received', () => {
      wrapper = shallow(<HyperText {...props} prefix="TEST_PREFIX" />);
      const textProps = wrapper
        .find('CustomText')
        .at(0)
        .props() as ICustomTextProps;
      expect(textProps.text).toEqual('TEST_PREFIX ');
      expect(textProps.color).toEqual(COLORS.GRAY_MEDIUM);
    });

    it('should render prefix text with "prefixColor" prop received', () => {
      wrapper = shallow(
        <HyperText
          {...props}
          prefix="TEST_PREFIX"
          prefixColor={COLORS.ALERT_LIGHT}
        />,
      );
      const textProps = wrapper
        .find('CustomText')
        .at(0)
        .props() as ICustomTextProps;
      expect(textProps.color).toEqual(COLORS.ALERT_LIGHT);
    });

    it('should render button with "customStyle" prop when received', () => {
      wrapper = shallow(<HyperText {...props} customStyles={{}} />);
      const buttonProps = wrapper
        .find('TouchableOpacity')
        .props() as TouchableOpacityProps;
      expect(buttonProps.style).toEqual({});
    });
  });
});
