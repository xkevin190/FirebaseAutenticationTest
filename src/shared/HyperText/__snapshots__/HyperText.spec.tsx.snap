// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HyperText (Unit-Test) general functionality should match snapshot 1`] = `
<TouchableOpacity
  onPress={[MockFunction]}
  testID="TEST_ID"
>
  <Text>
    <CustomText
      bodyType="paragraph-bold"
      color="#2162E3"
      text="TEST_TEXT"
    />
  </Text>
</TouchableOpacity>
`;
